arch/x86/core/thread.c

```diff
@@ -204,6 +204,9 @@ void _new_thread(struct k_thread *thread, char *pStackMem, size_t stackSize,
 
 	unsigned long *pInitialThread;
 
+#if _STACK_GUARD_SIZE
+	pStackMem += _STACK_GUARD_SIZE;
```

Comment:
You don't need the #if.  If _STACK_GUARD_SIZE is zero, then the += will optimize out.